R/expect-known.R

```diff
@@ -88,7 +88,7 @@ compare_file <- function(path, lines, ..., update = TRUE, info = NULL) {
     warning("Reference output is not UTF-8 encoded", call. = FALSE)
   }
 
-  comp <- waldo_compare(lines, enc2native(old_lines), ..., x_arg = "new", y_arg = "old")
+  comp <- waldo_compare(enc2utf8(lines), old_lines, ..., x_arg = "new", y_arg = "old")
```

Comment:
I think this misleading. I debugged this on Windows, and
* `lines` is apparently already marked as UTF-8 here, because `eval_with_output()` uses brio to read back the output, and brio marks it as UTF-8.
* So `enc2utf8()` does not actually do anything here.
* `lines` is not actually in UTF-8, but in the native encoding, because that's how Windows writes out the output.
* `old_lines` is also marked as UTF-8, because `brio::read_lines()` marks it so, but it is not UTF-8, it is in the native encoding.

The test case for this in `test-expect-known.R` succeeds because both `enc2utf8(lines)` and `old_lines` are in the native encoding and they are both marked as UTF-8.

If the contents of `lines` and/or `old_lines` is not compatible with UTF-8 (like in our own test case), then testthat will warn about the non-UTF-8 contents.